pytorch_lightning/callbacks/finetuning.py

```diff
@@ -82,6 +84,36 @@ def finetune_function(self, pl_module, current_epoch, optimizer, optimizer_idx):
                     )
     """
 
+    def __init__(self):
+        self._internal_state = {}
+
+    def on_save_checkpoint(
+        self,
+        trainer: 'pl.Trainer',
+        pl_module: LightningModule,
+        checkpoint: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        return self._internal_state
+
+    def on_load_checkpoint(
+        self, trainer: 'pl.Trainer', pl_module: LightningModule, callback_state: Dict[str, Any]
+    ) -> None:
+        self._internal_state = callback_state
+        # restore the param_groups created during the previous training.
+        _map_name_to_p = {n: p for n, p in pl_module.named_parameters()}
```

Comment:
Also, no point in prefixing local variables with underscore

Reply:
Also, maybe rename the variable to `named_parameters`?

Reply:
Done !